packages/mgt-components/src/components/mgt-todo/mgt-todo.ts

```diff
@@ -340,25 +358,46 @@ export class MgtTodo extends MgtTasksBase {
       `
       : html``;
 
-    const taskDeleteTemplate = this.readOnly
-      ? html``
-      : html`
-        <fluent-button class="task-delete"
-          @click="${() => this.removeTask(task.id)}"
-          aria-label="${this.strings.deleteTaskLabel}"
-        >
-          ${getSvg(SvgIcon.Delete)}
-        </fluent-button>
-      `;
+    const taskOptions = this.readOnly
+      ? nothing
+      : mgtHtml`
+          <mgt-dot-options
+            class="dot-options"
+            .options="${{
+              [this.strings.editTaskOption]: () => this.updateTask(task),
+              [this.strings.deleteTaskOption]: () => this.removeTask(task.id)
+            }}"
+          ></mgt-dot-options>`;
+
+    const taskDeleteTemplate = html`
+      <fluent-button class="task-delete"
+        @click="${() => this.removeTask(task.id)}"
+        aria-label="${this.strings.deleteTaskOption}">
+        ${getSvg(SvgIcon.Delete)}
+      </fluent-button>`;
 
     if (this.hasTemplate('task-details')) {
       taskDetailsTemplate = this.renderTemplate('task-details', context, `task-details-${task.id}`);
     } else {
+      const changeTaskDetailsTemplate = html`
+      <fluent-text-field 
+        autocomplete="off"
+        appearance="outline"
+        class="title"
+        id=${task.id}
+        .value="${task.title}"
+        aria-label="${this.strings.editTaskLabel}"
+        @keydown="${(e: KeyboardEvent) => this.handleChangeKeyDown(e, task)}"
```

Comment:
I think that the intent is to have the changes to the name save back to the server when the input loses focus or the user presses the enter key.

Given this I'd drop the change handler and add a blur handler.